File: app/src/main/java/com/vsa/filmoteca/view/MoviesListView.java
package com.vsa.filmoteca.view;

import com.vsa.filmoteca.view.adapter.EventDataProvider;
import com.vsa.filmoteca.view.dialog.interfaces.OkCancelDialogListener;

/**
 * Created by seldon on 10/03/15.
 */
public interface MoviesListView {

    void showChangeLog();

    void showProgressDialog();

    void hideProgressDialog();

    void stopRefreshing();

    void showTitle(int moviesCount);

    void showWifiRequestDialog(OkCancelDialogListener okCancelDialogListener);

    void showWifiSettings();

    void showTimeOutDialog();

    void showNoEventsDialog();

    void navigateToDetail(String url, String title, String date);

    void showAboutUs();

    void setMovies(EventDataProvider dataProvider);

    boolean isListLoaded();

    void finish();

}

File: app/src/main/java/com/vsa/filmoteca/presentation/view/MoviesListView.kt
package com.vsa.filmoteca.presentation.view

import com.vsa.filmoteca.core.presentation.view.LoadingView
import com.vsa.filmoteca.presentation.view.adapter.model.MovieViewModel
import com.vsa.filmoteca.presentation.view.dialog.interfaces.OkCancelDialogListener

/**
 * Created by seldon on 10/03/15.
 */
interface MoviesListView : LoadingView {

    fun showChangeLog()

    fun stopRefreshing()

    fun showTitle(moviesCount: Int)

    fun showWifiRequestDialog(okCancelDialogListener: OkCancelDialogListener)

    fun showWifiSettings()

    fun showTimeOutDialog()

    fun showNoEventsDialog()

    fun navigateToDetail(url: String, title: String, date: String)

    fun showAboutUs()

    fun setMovies(movies: List<MovieViewModel>)

    fun updateWidget()

    fun finish()

}
